app/views/posts/show.html.erb

```diff
@@ -0,0 +1,43 @@
+<div class="container mt-5">
+<h2>Post Details</h2>
+  <div class="d-flex align-items-start mb-4">
+    <div class="border p-3 flex-grow-1 d-flex flex-column">
+    <h1><%= @post.title %></h1>
+    <p>Written by: <%= @post.author.name %></p>
+    <p><%= @post.text %></p>
+      <div class="mt-auto text-right">
+        <small><%= @post.likes_counter %> likes</small>
+        <small><%= @post.comments_counter %> comments</small>
+      </div>
+    </div>
+  </div>
+  
+  <div class="card">
+    <div class="card-body">
+      <h4>Comments</h4>
+      <ul class="list-group">
+      <% @recent_comments.each do |comment| %>
+        <li class="list-group-item">
+          <strong><%= comment.author.name %></strong>: <%= comment.text %>
+          <% if can? :destroy, comment %>
+          <%= button_to 'Delete', user_post_comment_deletion_path(@post.author, @post, comment), method: :delete, data: { confirm: 'Are you sure?' } %>
+          <% end %>
+        </li>
+      <% end %>
+   </ul>   
+    </div>
+  </div>
+
+  <%= form_with model: [@post, Comment.new], url: user_post_comments_path(params[:user_id], @post), local: true do |form| %>
+    <%= form.label :text, "Add a Comment:" %>
+    <%= form.text_area :text %>
+    <%= form.submit "Comment" %>
+  <% end %>
+
+  <div class="mt-3">
+    <% if can? :destroy, @post %>
```

Comment:
- [ ] Hello, when i try to delete a post I run into below error and the reason is you did not create ability file

![Screenshot from 2023-12-21 21-40-24](https://github.com/EvansSnave/Blog_app_rails/assets/112261237/da67e29a-f185-44f8-a6ce-727297588a9d)

after adding cancancan gem file, follow below steps

1. run `bundle install`
2. run `rails g cancan:ability` to generate user permissions which are defined in `Ability` class and define abilities, it should be something like below

```ruby
 # app/models/ability.rb

 class Ability
 include CanCan::Ability

     def initialize(user)

         user ||= User.new # guest user (not logged in)

         if user.admin? 
         can :manage, :all
         else
         can :manage, Post, author: user  # owner of post can perform all view, update or destroy own post if Post.author == user
         can :manage, Comment, author: user
         can :read, :all
         end
     end
 end
```

3. add migration to add role to users table
4. Uses **CanCanCan** for Post authorization in `Post Controller`, see below as example code

```ruby
  class PostsController < ApplicationController
    load_and_authorize_resource
  end
```